### PR TITLE
feat(kubernetes/artifacts): Support deploying images dynamically

### PR DESCRIPTION
**Pull request by lwander**

### CODE REVIEW

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/artifact/ArtifactReplacer.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    replacers.add(replacer);
+    return this;
+  }
+
+  public KubernetesManifest replaceAll(KubernetesManifest input, List<Artifact> artifacts) {
+    DocumentContext document;
+    try {
+      document = JsonPath.using(configuration).parse(mapper.writeValueAsString(input));
+    } catch (JsonProcessingException e) {
+      log.error("Malformed manifest - not sure how this is possible", e);
```
Comment (jtk54) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/artifact/ArtifactReplacer.java]: > not sure how this is possible

Remove this and replace with an explanation of why this is impossible.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/artifact/ArtifactReplacer.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      throw new RuntimeException(e);
+    }
+
+    replacers.forEach(r -> artifacts.forEach(a -> r.replace(document, a)));
+
+    try {
+      return mapper.readValue(document.jsonString(), KubernetesManifest.class);
+    } catch (IOException e) {
+      log.error("Malformed Document Context - not sure how this is possible", e);
```
Comment (jtk54) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/artifact/ArtifactReplacer.java]: Same here.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/artifact/ArtifactReplacer.java (hunk 1 above)
Comment (lwander) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/artifact/ArtifactReplacer.java]: An explanation is overkill - should be clear from the line above (reading json as json). But I'll drop it from the message.

**Comment by lwander** [comment]
https://github.com/spinnaker/spinnaker/issues/2158

**Pull request merged**